bootstrap.sh

```diff
@@ -0,0 +1,267 @@
+
+    echothis "$(timestamp) [2-4]: Generate terraform plan..."
+    if [ -f "${PWD}/$TF_PLAN" ]
+    then
+        echothis "Found existing plan! Replacing..."
+        rm $TF_PLAN
+    fi
+    
+    terraform plan -out=$TF_PLAN
+    echothis $STEP_SUCCESS; sleep 1
+
+    echothis "$(timestamp) [2-5]: Instruction - Review the generated plan before deploying. Run 'terraform show bootstrap.tfplan'.\n"; sleep 2
+
+    pause "[CHECKPOINT #2]: Do you want to continue? [Y/N]: " # CHECKPOINT!!!!
+
+# >>>> STEP 3 - Deploy bootstrap resources <<<<<<<
+    echothis "\n>>>>>>>>>> RUNNING: STEP #3.\n\n"
+    echothis "$(timestamp) [3-1]: Deploying the bootstrap infrastructure..."
+    
+    terraform apply -auto-approve $TF_PLAN
+    
+    echothis $STEP_SUCCESS; sleep 1
+    echothis "$(timestamp) [3-2]: Instruction - Review terraform output by running 'terraform show output'.\n"; sleep 2
+    pause "[CHECKPOINT #3]: Do you want to continue? [Y/N]: " # CHECKPOINT!!!!
+
+# >>>> STEP 4 - Update and migrate terraform state to GCS backend <<<<
+    echothis "\n>>>>>>>>>> RUNNING: STEP #4.\n\n"    
+    echothis "$(timestamp) [4-1]: Updating terraform gcs backend conf with state bucket name..."
+
+    export BUCKET_NAME=$(terraform output -raw tf-state-bucket-name)
+    #export BUCKET_NAME="appdev-320214-terraform"
+    
+    if [ -z "${BUCKET_NAME}" ] # Check variable is populated with bucket name from terraform output
+    then
+        echothis "$(timestamp) [ERROR]: GCS bucket name not populated.\n"
+        exit 1
+    fi
+
+    # Preserve example file for templated use by copying it before replacing the embedded placeholder
+    cp $WORKDIR/backend.tf.example $WORKDIR/backend.tf.stage
+    sed -i "s/BUCKET_NAME_PLACEHOLDER/${BUCKET_NAME}/" $WORKDIR/backend.tf.stage
+    mv $WORKDIR/backend.tf.stage $WORKDIR/backend.tf
+
+    if [ -f "$WORKDIR/backend.tf" ]; then
+        echothis $STEP_SUCCESS; sleep 1
+    else
+        echothis "$(timestamp) [ERROR]: Backend.tf file not generated successfully.\n"
+        exit 1
+    fi
+
+    echothis "$(timestamp) [4-2]: Instruction - Review backend config file below before proceeding to the next step:\n\n"
+    
+    cat ${PWD}/backend.tf | tee -a $LOG_FILE_PATH >&3
+    
+    echothis "\n"; sleep 1
+    pause "[CHECKPOINT #4]: Do you want to continue? [Y/N]: " # CHECKPOINT!!!!
+
+    echothis "$(timestamp) [4-3]: Migrating terraform state from local to GCS remote..."
+    terraform init -migrate-state -force-copy | tee -a $LOG_FILE_PATH >&3
+    echothis $STEP_SUCCESS; sleep 1
+
+# >>>> STEP 5 - Generate github keys and configure git global <<<<
+    echothis "\n>>>>>>>>>> RUNNING: STEP #5.\n\n"
+
+    echothis "$(timestamp) [5-1]: Generate keys to connect to private Github repo..."
+    ssh-keygen -t rsa -b 4096 -N "" -q -C "${GITHUB_BOT_USER}" -f ~/.ssh/id_github 2>&1> /dev/null # suppress output
+    ssh-keyscan -t rsa github.com 2>&1 | tee ~/.ssh/known_hosts > /dev/null 2&>1 # suppress output
```

Comment:
Would be good to add a check here to confirm if these ssh key files are already present. Otherwise if present, the scripts fails

Reply:
Scripts amended as advised. Checks included in function "step5_generate_sshkeys_and_configure_git".